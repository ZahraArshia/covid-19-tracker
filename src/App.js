import { Routes, Route } from 'react-router-dom';
import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import Navigation from './components/Navigator';
import HomePage from './components/HomePage';
import DetailsPage from './components/DetailsPage';
import { getCountries } from './redux/modules/country';

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    getCountries()(dispatch);
  }, []);
  return (
    <div>
      <Navigation />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/:id" element={<DetailsPage />} />
      </Routes>
    </div>

  );
}

export default App;
