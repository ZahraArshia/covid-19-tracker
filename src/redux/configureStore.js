import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import countriesReducer from './modules/country';

const reducer = combineReducers({
  countriesReducer,
});

const configureStore = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default configureStore;
