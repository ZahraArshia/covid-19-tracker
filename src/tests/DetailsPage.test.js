/* eslint-disable camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from './mocks/configureStore-mock';
import App from '../App';

describe('Countries snapshot', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={configureStore}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
