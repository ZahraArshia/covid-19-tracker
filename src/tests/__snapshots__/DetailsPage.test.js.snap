// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot renders correctly 1`] = `
<div>
  <nav>
    <section
      className="navLinks"
    >
      <div>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="back"
            src="https://img.icons8.com/ios-filled/24/ffffff/back.png"
          />
        </a>
      </div>
      <div
        className="mainTitle"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Covid-19 Tracker
        </a>
      </div>
      <div
        className="navIcons"
      >
        <img
          alt="mic"
          className="navIcon"
          src="https://img.icons8.com/fluency-systems-filled/24/ffffff/microphone.png"
        />
        <img
          alt="settings"
          className="navIcon"
          src="https://img.icons8.com/external-tanah-basah-glyph-tanah-basah/24/ffffff/external-setting-essentials-pack-tanah-basah-glyph-tanah-basah.png"
        />
      </div>
    </section>
  </nav>
  <div>
    <header
      className="cover"
    >
      <div
        className="title"
      >
        <div
          className="logo"
        >
          <img
            alt="virus-transmission"
            src="https://img.icons8.com/external-wanicon-lineal-wanicon/120/ffffff/external-world-virus-transmission-wanicon-lineal-wanicon.png"
          />
          <div>
            <h1>
              2022
            </h1>
            <h3>
              covid-19 world tracker
            </h3>
          </div>
        </div>
      </div>
    </header>
    <div
      className="searchBar"
    >
      <div
        className="dateSelection"
      >
        <h2>
           Stats by country
        </h2>
        <h3
          className="date"
        >
          2022-04-21
        </h3>
      </div>
      <input
        className="searchField"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search Countries"
        type="text"
        value=""
      />
    </div>
    <ul
      className="Countriescontainer"
    />
  </div>
</div>
`;
